pkg/builder/parameters.go

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package builder
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"sort"
+	"strconv"
+	"strings"
+
+	"k8s.io/kube-openapi/pkg/validation/spec"
+)
+
+// collectSharedParameters finds those parameters that show up often and hence can be
+// shared across all the paths to save space.
+func collectSharedParameters(sp *spec.Swagger) (namesByJSON map[string]string, ret map[string]spec.Parameter, err error) {
+
+	if sp == nil || sp.Paths == nil {
+		return nil, nil, nil
+	}
+
+	countsByJSON := map[string]int{}
+	shared := map[string]spec.Parameter{}
+	var keys []string
+
+	collect := func(p *spec.Parameter) error {
+		if (p.In == "query" || p.In == "path") && p.Name == "name" {
+			return nil // ignore name parameter as they are never shared with the Kind in the description
+		}
+		if p.In == "body" && p.Name == "body" && !strings.HasPrefix(p.Ref.String(), "#/definitions/io.k8s.apimachinery") {
+			return nil // ignore body parameter as they are never shared with the custom schema for each kind
+		}
+
+		bs, err := json.Marshal(p)
+		if err != nil {
+			return err
+		}
+
+		countsByJSON[string(bs)]++
```

Comment:
Should we save `string(bs)` somewhere or is this not doing a copy every single time (since you do `string(bs)` 4 times in 4 lines)?

Reply:
good question. Never thought about it. I thought this is just a noop on a slice.

Reply:
GPT says it's a copy 😁, i.e. O(n).

Reply:
done